code_challenges/hacker_rank/designer_door_mat.py

```diff
@@ -0,0 +1,55 @@
+def draw_door_mat(size, greeting):
+    height = int(size[0])
+    length = int(size[1])
+    starter = int((length - 1) / 2)
+    greeting_design_count = int((length - int(len(greeting))) / 2)
+    detail_a = ".|."
+    detail_b = "-"
+    result = []
+    detail_b_count = int((length - 1) / 2)
+    detail_a_count = 1
+
+    if height % 2 != 0:
```

Comment:
Here instead of nesting all the logic inside the `guard clause`. This should probably be like this for readability:

```
    if height % 2 == 0:
        return (
            "Mat size must be N * M . (N is an odd natural number, and M is 3 times N."
        )
        
    for n in range((int(height / 2))):
        result.append(.....
````

[Reading on guard clauses.](https://medium.com/lemon-code/guard-clauses-3bc0cd96a2d3)